### PR TITLE
Adds the use of delta-time to the contribution guidelines

### PR DESCRIPTION
**Pull request by ghost**
Delta-time is a concept we added but isn't used a lot yet, I think adding it to the contribution guidelines with an explanation of how to actually use it (As its a concept that only really applies to game dev afaik, and isn't something everyone is familiar with) will increase the amount of people using it.

If there's any issues with my explanation feel free to comment and I'll update it.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -99,6 +99,65 @@ The previous code made compliant:
...
+
+As the mobs subsystem runs once every 2 seconds, the mob now loses 4 health every process, or 2 health per second. This is called frame dependent programming.
+
+Why is this an issue? If someone decides to make it so the mobs subsystem processes once every second (2 times as fast), your effects in process() will also be two times as fast. Resulting in 4 health loss per second rather than 2.
+
+How do we solve this? By using delta-time. Delta-time is the amount of seconds you would theoretically have between 2 process() calls. In the case of the mobs subsystem, this would be 2 (As there is 2 seconds between every call in process()). Here is a new example using delta-time
+
+```DM
+/mob/testmob
+	var/health = 2
```
Comment (Ghilker) [inline .github/CONTRIBUTING.md:124]: var/health = 100

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -99,6 +99,39 @@ The previous code made compliant:
...
+### All process procs need to make use of delta-time and be frame independent
+
+In a lot of our older code, process() is frame dependent. As an example, here's some example mob code.
+
+```DM
+/mob/testmob
+	var/health = 100
+	var/health_loss = 4 //We want to lose 2 health per second, so 4 per SSmobs process
+
+/mob/testmob/process(delta_time) //SSmobs runs once every 2 seconds
```
Comment (TheFakeElon) [inline .github/CONTRIBUTING.md:111]: Isn't this meant to be the one with our delta_time

Comment (Qustinnus) [reply .github/CONTRIBUTING.md:111]: it's a default arg on process(), It's just not being utilized in this example.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -99,6 +99,39 @@ The previous code made compliant:
...
+```DM
+/mob/testmob
+	var/health = 100
+	var/health_loss = 2 //Health loss every second
+
+/mob/testmob/process(delta_time) //SSmobs runs once every 2 seconds
+	health -= health_loss * delta_time
+
+```
+In the above example, we made our health_loss variable a per second value rather than per process. In the actual process() proc we then make use of deltatime. Because SSmobs runs once every  2 seconds. Delta_time would have a value of 2. This means that by doing health_loss * delta_time, you end up with the correct amount of health_loss per process, but if for some reason the SSmobs subsystem gets changed to be faster or slower in a PR, your health_loss variable will work the same.
```
Comment (ATH1909) [inline .github/CONTRIBUTING.md:131]: you should probably also mention that this makes the code more readable for people who don't know exactly how often process() is called for mobs, objects, etc.

Comment (ATH1909) [reply .github/CONTRIBUTING.md:131]: like, don't we have some stuff that processes once every second, but some other stuff that processes every 2 seconds?

**Comment by ExcessiveUseOfCobblestone** [comment]
@tgstation/commit-access

**Comment by Jared-Fogle** [comment]
i like it and will merge later if nobody has objections

**Comment by SpaceManiac** [comment]
We should consider explicitly saying not to do for-loops based on the delta time

**Comment by SpaceManiac** [comment]
also your commit author info is broken to hell

**Comment by LemonInTheDark** [comment]
Simple enough for me to understand, should be just fine.
Might be worth mentioning when it's appropriate to use, and when it's not, or at least making it clear that there is a choice.

**Review by spookydonut — APPROVED** [review]

**Comment by spookydonut** [comment]
> 
> 
> We should consider explicitly saying not to do for-loops based on the delta time

what?

**Comment by Cyberboss** [comment]
```dm
for(var/I in 1 to delta_time / 2)
    (do interval shit)
```

**Comment by Qustinnus** [comment]
what?

**Comment by SpaceManiac** [comment]
having a for-loop that iterates more when the delta-time is larger breaks the entire purpose of the delta-time being configurable in the first place, which is so that the subsystem can be made to do less total work

all your commits are authored by `ff <ff>` instead of a sensible email; please read https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-in-git
